### DISCUSSION TITLE
Hi Rector Team: One question How would you deduplicate phpdoc blocks using rector?

### DISCUSSION THREAD

**Discussion by alexandreelise** [rectorphp/rector#7724] [Q&A] [answered]
How to remove unexpected and useless php doc blocks that are sometimes generated through IDE many times again and again. Can it be fixed using a Rector? If not thanks anyway. Have a great day and keep on coding good stuff.

For example something like that

```php
/**
* @var int $a
*/
/**
* @var int $a
*/
public function show($a) 
{
//...some code goes here
}

```

**Comment by alexandreelise** [comment]
Hi, should I rephrase my question or refine it? Is it too poor quality or off-topic to be answered? Please shed a light to let me understand what did I do wrong.

**Reply by alexandreelise** [reply]
@samsonasik can you please let me know what did I do wrong so I won't do it again?

**Comment by TomasVotruba** [comment, marked as answer]
Hi, you can write a custom rule that checks "comments" attribute of the `ClassMethod` node


Here is how to write a custom rule:
https://getrector.com/documentation/custom-rule

**Reply by alexandreelise** [reply]
Thanks
